client/src/components/GameEnv/3DEnv/3DEnv.jsx

```diff
@@ -52,21 +56,76 @@ function ArWing() {
   );
 }
 
-const ThreeDEnv = () => (
+function Cards({ position, slot }) {
+  const group = useRef();
+  const { nodes } = useLoader(GLTFLoader, card);
+  const [clicked, setClicker] = useState(false);
+  // useFrame will run outside of react in animation frames to optimize updates.
+  useFrame(() => {
+    group.current.rotation.x = 0.9;
+  });
+
+  const handleClick = () => setClicker(!clicked);
+
+  const prop = {
+    position: [...position],
+  };
+
+  prop.position[2] = -10;
+
+  if (slot) {
+    position = [0, 0, 100];
+  }
+
+  return (
+    // Add a ref to the group. This gives us a hook to
+    // manipulate the properties of this geometry in the useFrame callback.
+    <group ref={group} position={clicked ? prop.position : position}>
+      <mesh
+        visible
+        geometry={nodes.mesh_0.geometry}
+        scale={clicked ? new THREE.Vector3(0.04, 0.04, 0.04) : new THREE.Vector3(0.02, 0.02, 0.02)}
+        onClick={handleClick}
+      >
+        <meshPhongMaterial attach="material" color="gold" />
+      </mesh>
+    </group>
+  );
+}
+Cards.propTypes = {
+  position: PropTypes.arrayOf(PropTypes.number).isRequired,
+  slot: PropTypes.bool.isRequired,
+};
+
+const ThreeDEnv = ({ slots }) => (
   <div>
     <div style={{ height: window.innerHeight * 0.73 }}>
       <Canvas>
         <color attach="background" args={['gray']} />
         <OrbitControls />
         <directionalLight intensity={0.5} />
         <ambientLight intensity={0.5} />
-        <spotLight position={[10, 15, 10]} angle={0.9} />
+        <spotLight position={[20, 20, 10]} angle={0.9} />
         <Suspense fallback={<Loading />}>
-          <ArWing />
+          <Table />
+          <Loading />
+          <Cards position={[6, 2, -13]} slot={slots[3]} />
```

Comment:
are these cards hardcoded in?

Reply:
not really its where the cards will be placed on the table